vscode/src/test/suite/testutils.ts

```diff
@@ -192,3 +192,278 @@ export async function dumpJava(): Promise<void> {
+        console.log("Keys don't match");
+        return false;
+    }
+
+    var obj1 = JSON.parse(fs.readFileSync(jsonFileBase, 'utf8'));
+    var obj2 = JSON.parse(fs.readFileSync(jsonFileTarget, 'utf8'));
+    var keys = new Set(Object.keys(obj1));
+    var matched: boolean = true;
+    for (const key of keys) {
+
+        if (matchTemplate(obj1[key], obj2[key])) {
+            continue;
+        }
+        console.log("Templates of the key " + key + " don't match .'" + obj1[key] + "' and '" + obj2[key] + "' don't match");
+        matched = false;
+    }
+    return matched;
+}
+
+
+
+export function matchKeys(jsonFilePath1: string, jsonFilePath2: string): boolean {
+    return setEqual(getKeysFromJSON(jsonFilePath1), getKeysFromJSON(jsonFilePath2));
+}
+
+/**
+ * 
+ * @param dirPath 
+ * @param ignoredDirEntriesNames 
+ * @param validKeyValues 
+ * @returns number of js files in the directory specified which invoke l10n.value with incorrect key or placeholder map 
+ */
+export function checkl10nUsageInFiles(dirPath: string, ignoredDirEntriesNames: Set<string>, validKeyValues: any): number {
+
+    let result: number = 0;
+    fs.readdirSync(dirPath).forEach(DirEntryName => {
+        if (ignoredDirEntriesNames.has(DirEntryName)) return;
+        const absPath = path.join(dirPath, DirEntryName);
+        const stats = fs.lstatSync(absPath);
+        if (stats.isFile() && path.extname(DirEntryName) === ".js") {
+            if (
+                !(checkl10UsageInFile(absPath, /l10n.value\("([^"]*)"\)/g, validKeyValues) &&
+                    checkl10UsageInFile(absPath, /l10n.value\('([^']*)'\)/g, validKeyValues) &&
+                    checkl10UsageInFile(absPath, /l10n.value\('([^']*)',\s*\{([^\}]*)\}\s*\)/g, validKeyValues) &&
+                    checkl10UsageInFile(absPath, /l10n.value\("([^"]*)",\s*\{([^\}]*)\}\s*\)/g, validKeyValues)
+                )
+            ) result++;
+        } else if (stats.isDirectory()) {
+            result += checkl10nUsageInFiles(absPath, ignoredDirEntriesNames, validKeyValues);
+        }
+    });
+    return result;
+}
+
+
+export function checkConfigurationLocalisation(configuration: any, validKeys: Set<string>): boolean {
+    let localized: boolean = true;
+    const configPropertiesIds = Object.keys(configuration.properties);
+    const propertiesLocalisableFields = ["description", "enumDescriptions"];
+    let property: any;
+    for (const propertyId of configPropertiesIds) {
+        property = configuration.properties[propertyId];
+        if (!isLocalizedObj(property, propertiesLocalisableFields, propertyId, "Configuration Property", validKeys)) localized = false;
+    }
+    return localized;
+}
+
+export function checkDebuggersLocalisation(debuggers: any, validKeys: Set<string>): boolean {
+    let localized = true;
+    const propertiesLocalisableFields = ['description'];
+    const configLocalisableFields = ['name'];
+    const snippetLocalisableFields = ['label', 'description'];
```

Comment:
If available, would you please add the URL of the spec document which defines these as a comment here?

Reply:
It's not available to my knowledge.